repository.go

```diff
@@ -1011,3 +1013,77 @@ func (r *Repository) ResolveRevision(rev plumbing.Revision) (*plumbing.Hash, err
 
 	return &commit.Hash, nil
 }
+
+type RepackConfig struct {
+	// StatusChan for status updates, may be nil.
+	StatusChan plumbing.StatusChan
+	// UseRefDeltas configures whether packfile encoder will use reference deltas.
+	// By default OFSDeltaObject is used.
+	UseRefDeltas bool
+	// PackWindow for packing objects.
+	PackWindow uint
```

Comment:
Sorry for not being clear, I meant that it should use `r.Storer.Config().Pack.Window`, rather than having a new one passed in.  This way works for us, but I'm not sure they'd like it upstream.